src/error.rs

```diff
@@ -11,48 +10,66 @@ pub enum DeserializeError {
     #[error("Invalid header: {0}")]
     InvalidHeader(String),
     /// If a value has a size that was unexpected, e.g. UInt32Property has 8 bytes size
-    #[error("Invalid value size, expected {0} got {1} at position {2}")]
+    #[error("Invalid value size, expected {0} got {1} at position {2:#x}")]
     InvalidValueSize(u64, u64, u64),
     /// If a string has invalid size
-    #[error("Invalid string size, got {0} at position {1}")]
+    #[error("Invalid string size {0} at position {1:#x}")]
     InvalidString(u32, u64),
     /// If a boolean has invalid value
-    #[error("Invalid boolean value, got {0} at position {1}")]
+    #[error("Invalid boolean value {0} at position {1:#x}")]
     InvalidBoolean(u32, u64),
     /// If a hint is missing.
-    #[error("Missing hint for struct {0} at path {1}, cursor position: {2}")]
+    #[error("Missing hint for struct {0} at path {1} at position {2:#x}")]
     MissingHint(String, String, u64),
     /// If an argument is missing
-    #[error("Missing argument: {0} at position {1}")]
+    #[error("Missing argument: {0} at position {1:#x}")]
     MissingArgument(String, u64),
-    /// If an EnumProperty has an invalid enum type
-    #[error("Invalid enum type {0} at position {1}")]
-    InvalidEnumType(String, u64),
     /// If a Property creation fails
-    #[error("Invalid property {0} at position {1}")]
+    #[error("Invalid property {0} at position {1:#x}")]
     InvalidProperty(String, u64),
     /// Invalid enum value
-    #[error("{0}")]
-    InvalidEnumValue(Box<str>),
-}
-
-impl<T: TryFromPrimitive> From<TryFromPrimitiveError<T>> for DeserializeError {
-    fn from(value: TryFromPrimitiveError<T>) -> Self {
-        DeserializeError::InvalidEnumValue(value.to_string().into_boxed_str())
-    }
+    #[error("No discriminant in enum `{0}` matches the value `{1}` at position {2:#x}")]
+    InvalidEnumValue(String, i8, u64),
```

Comment:
shouldn't errors have box<str>?

Reply:
I don't have a strong opinion one way or the other. This code was not boxing the other Strings in the DeserializeError enum, and I wanted to make it consistent. I'd be fine with converting them all to `Box<str>` to signal immutability.